models/lobby/server.go

```diff
@@ -0,0 +1,255 @@
+	LobbyId int
+
+	Players        []TF2RconWrapper.Player // current number of players in the server
+	AllowedPlayers []TF2RconWrapper.Player
+
+	Config     *ServerConfig // config that should run before the lobby starts
+	WhiteList  string        // whitelist that should run before the lobby starts
+	MaxPlayers int
+
+	// timer that will verify()
+	Ticker verifyTicker
+
+	//ChatListener  *TF2RconWrapper.RconChatListener
+	RconPassword  string // will store the rcon password specified by the client
+	LobbyPassword string // will store the new server password from the lobby
+}
+
+// timer used in verify()
+type verifyTicker struct {
+	Ticker *time.Ticker
+	Quit   chan struct{}
+}
+
+func (t *verifyTicker) Close() {
+	close(t.Quit)
+}
+
+type Map struct {
+	Name   string
+	Config string
+	League string
+	Mode   string
+}
+
+func NewServer() *Server {
+	return new(Server)
+}
+
+// after create the server var, you should run this
+//
+// things that needs to be specified before run this:
+// -> Map
+// -> Mode
+// -> League
+// -> LobbyId
+// -> Address
+// -> RconPassword
+// -> LobbyPassword
+//
+func (s *Server) Setup() error {
+	fmt.Println("[Server.Setup]: Setting up server -> [" + s.Address + "] from lobby [#" + strconv.Itoa(s.LobbyId) + "]")
+
+	// connect to rcon
+	var err error
+	s.Rcon, err = TF2RconWrapper.NewTF2RconConnection(s.Address, s.RconPassword)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// changing server password
+	passErr := s.ChangePassword(s.LobbyPassword)
+
+	if passErr != nil {
+		log.Fatal(passErr)
+	}
+
+	// kick players
+	fmt.Println("[Server.Setup]: Connected to server, getting players...")
+	kickErr := s.KickAll()
+
+	if kickErr != nil {
+		log.Fatal(kickErr)
+	} else {
+		fmt.Println("[Server.Setup]: Players kicked, running config!")
+	}
+
+	// run config
+	config := NewServerConfig()
+	config.League = s.League
+	config.Type = s.Type
+	config.Map = s.Map
+	cfg, cfgErr := config.Get()
+
+	if cfgErr == nil {
+		config.Data = cfg
+		configErr := s.ExecConfig(config)
+
+		if configErr != nil {
+			log.Fatal(configErr)
+		}
+	} else {
+		log.Fatal(cfgErr)
+	}
+
+	// change map
+	mapErr := s.ChangeMap(s.Map)
+
+	if mapErr != nil {
+		log.Fatal(mapErr)
+	}
+
+	// verify's timer
+	s.Ticker.Ticker = time.NewTicker(10 * time.Second)
+	s.Ticker.Quit = make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-s.Ticker.Ticker.C:
+				s.Verify()
+			case <-s.Ticker.Quit:
+				s.Ticker.Ticker.Stop()
+				return
+			}
+		}
+	}()
+
+	return nil
+}
+
+// runs each 10 sec
+func (s *Server) Verify() {
+	fmt.Println("[Server.Verify]: Verifing server -> [" + s.Address + "] from lobby [#" + strconv.Itoa(s.LobbyId) + "]")
+
+	// check if all players in server are in lobby
+	s.Players = s.Rcon.GetPlayers()
+	for i := range s.Players {
+		// check if player is not in lobby but is in server
+		// ignores BOT (SourceTV)
+		if s.Players[i].SteamID != "BOT" && s.IsPlayerInLobby(s.Players[i].SteamID) == false {
```

Comment:
line 143: `bson.ObjectId` is a hex string, IIRC. So, if we want to expose that to the client, we would have to decide between using the hex string, or its numeral representation.